metrics/db_file_metrics.go

```diff
@@ -0,0 +1,99 @@
+package metrics
+
+import (
+	"sort"
+	"sync"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+)
+
+// FileMetric data structure for one file, will be maintained in a map
+type FileMetric struct {
+	ValidEntries   int32
+	InvalidEntries int32
+	ValidBytes     int64
+	InvalidBytes   int64
+}
+
+var (
+	once        sync.Once
+	fileMetrics map[int32]*atomic.Value // a map holding all the metrics for all files
+	lock        sync.Mutex
+)
+
+var ErrFileMetricNotExists = errors.New("fileMetric not exits")
+
+// Init called when start or restart, only the first call will take effect.
+func Init() {
+	once.Do(func() {
+		fileMetrics = make(map[int32]*atomic.Value)
+	})
+}
+
+// DeleteFileMetric delete a metric for a file, should be called when a file is deleted
+func DeleteFileMetric(fd int32) {
+	delete(fileMetrics, fd)
+}
+
+// UpdateFileMetric
+// for an existing fd, you should start with a new FileMetric{0,0,0,0},
+// then use it to accumulate the metric updates when you do your business,
+// after committing, update it into the fileMetrics using this method.
+func UpdateFileMetric(fd int32, delta *FileMetric) error {
```

Comment:
for the param name "fd", I think the better name is "fid". That will be more accurate and will not confuse people who reader this code.